packages/connect/src/protocol-connect/handler-factory.spec.ts

```diff
@@ -51,51 +53,89 @@ describe("createHandlerFactory()", function () {
     },
   } as const;
 
-  function stub<M extends MethodInfo>(
-    opt: {
-      service?: ServiceType;
-      method?: M;
-      impl?: MethodImpl<M>;
-    } & Partial<UniversalHandlerOptions>
+  function setupTestHandler<M extends MethodInfo>(
+    method: M,
+    opt: Partial<UniversalHandlerOptions>,
+    impl: MethodImpl<M>
   ) {
-    const method = opt.method ?? testService.methods.foo;
-    let implDefault: MethodImpl<M>;
-    switch (method.kind) {
-      case MethodKind.Unary:
-        // eslint-disable-next-line @typescript-eslint/require-await
-        implDefault = async function (req: Message, ctx: HandlerContext) {
-          ctx.responseHeader.set("stub-handler", "1");
-          return new ctx.method.O();
-        } as unknown as MethodImpl<M>;
-        break;
-      case MethodKind.ServerStreaming:
-        // eslint-disable-next-line @typescript-eslint/require-await
-        implDefault = async function* (req: Message, ctx: HandlerContext) {
-          ctx.responseHeader.set("stub-handler", "1");
-          yield new ctx.method.O();
-        } as unknown as MethodImpl<M>;
-        break;
-      case MethodKind.ClientStreaming:
-      case MethodKind.BiDiStreaming:
-        implDefault = (() => {
-          throw new Error("not implemented");
-        }) as unknown as MethodImpl<M>;
-        break;
-    }
-    const spec = createMethodImplSpec(
-      opt.service ?? testService,
-      method,
-      opt.impl ?? implDefault
+    const h = createHandlerFactory(opt)(
+      createMethodImplSpec(testService, method, impl)
     );
-    const f = createHandlerFactory(opt);
-    return f(spec);
+    const t = createTransport({
+      httpClient: createUniversalHandlerClient([h]),
+      baseUrl: "https://example.com",
+      readMaxBytes: 0xffffff,
+      writeMaxBytes: 0xffffff,
+      compressMinBytes: 0xffffff,
+      useBinaryFormat: true,
+      interceptors: [],
+      acceptCompression: [],
+      sendCompression: null,
+    });
+    return {
+      service: testService,
+      method: method,
+      handler: h,
+      transport: t,
+    };
   }
 
+  describe("returned handler", function () {
+    it("should surface headers for unary", async function () {
+      const { transport, service, method } = setupTestHandler(
+        testService.methods.foo,
+        {},
+        (req, ctx) => {
+          ctx.responseHeader.set("implementation-called", "yes");
+          return { value: req.value.toString(10) };
```

Comment:
question: just for my own learning, I'm curious what value it brings to do `toString(10)` when the default is already 10.  My first reaction is that it's worth it to be explicit because there are other situations where we use other radices, but I searched the codebase and didn't see any others.   just curious.

Reply:
Just to be explicit, but I wouldn't say it's the most important thing in the world. There should be a couple of other places where it's used.